### PR TITLE
Implement config_ordinal consistently for built in sources

### LINKED ISSUE #14377: Server.xml config sources do not respect config_ordinal
**Describe the bug**
The spec allows the ordinal of a built in config source to be changed by defining a property named `config_ordinal` which is read from that config source
> Note that a special property config_ordinal can be set within any built-in ConfigSource implementation. The default implementation of getOrdinal() will attempt to read this value. If found and a valid integer, the value will be used. Otherwise the respective default value will be used.

However, our built in config sources for server.xml variables and `<appProperties>` do not change their ordinal if a property with `config_ordinal` is set.

### PR DESCRIPTION
**Pull request by Azquelt**
Currently some of our config sources respect `config_ordinal` and some don't and each source has its own logic to look up this variable.

Make the code that does this common and use it for all built in config sources.

Fixes #14377

### CODE REVIEW

**Comment by Azquelt** [comment]
#build
#run-libby-bot

File: dev/com.ibm.ws.microprofile.config.1.1/src/com/ibm/ws/microprofile/config/sources/EnvConfigSource.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2017, 2018 IBM Corporation and others.
+ * Copyright (c) 2017, 2020 IBM Corporation and others.
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.microprofile.config.1.1/src/com/ibm/ws/microprofile/config/sources/EnvConfigSource.java:2]: 2018, 2020

Comment (Emily-Jiang) [reply dev/com.ibm.ws.microprofile.config.1.1/src/com/ibm/ws/microprofile/config/sources/EnvConfigSource.java:2]: ignore my last comment. I missed the 2017, bit when I first reviewed.

**Review by Emily-Jiang — APPROVED** [review]

File: dev/com.ibm.ws.microprofile.config.1.1/src/com/ibm/ws/microprofile/config/sources/EnvConfigSource.java (hunk 1 above)
Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.config.1.1/src/com/ibm/ws/microprofile/config/sources/EnvConfigSource.java:2]: I think this change is correct, 2017 is the original year (which stays the same) while 2020 is the year of the most recent change.

**Pull request merged**